app/views/admin/order_cycles/_filters.html.haml

```diff
@@ -6,7 +6,7 @@
   .filter_select.four.columns
     %label{ :for => 'involving_filter' }=t('.involving')
     %br
-    %input.ofn-select2.fullwidth{ :id => 'involving_filter', type: 'number', blank: "{id: 0, name: 'Any Enterprise'}", data: 'enterprises', ng: { model: 'involvingFilter' } }
+    %input.ofn-select2.fullwidth{ :id => 'involving_filter', type: 'number', blank: "{id: 0, name: '#{j t(".any_enterprise")}'}", data: 'enterprises', ng: { model: 'involvingFilter' } }
```

Comment:
That j in #{j is a typo or am I about to learn something I didnt know?

Reply:
No idea either

Reply:
would you mind replacing that `:id` has rocket for consistency?

Reply:
amazing, I'm learning!
Are you saying that we should use this j everywhere in all rails translations that are used by some form of js code in the process? 
We should document this in i18n wiki page.

Reply:
@sauloperez I didn't notice that it was using the two hash syntaxes in the same line! Thanks, I appended a commit replacing this.

Reply:
ping :-)
Can you please clarify "Are you saying that we should use this j everywhere in all rails translations that are used by some form of js code in the process?"

Reply:
So, while starting to make changes to the wiki, I realized that I could use the Angular filter `t` here instead. I revised this PR. (There are now no more conflicts with #2659, so this branch no longer includes the changes there.)

The I18n wiki page already says to prefer this filter, so I did not edit that. I updated the Code Conventions doc though to remind developers to [use `j` when inserting an arbitrary Ruby string to a JS or Angular expression](https://github.com/openfoodfoundation/openfoodnetwork/wiki/Code-Conventions#use-j-when-inserting-an-arbitrary-ruby-string-in-a-js-or-angular-expression).

Requesting another review, sorry!

Reply:
awesome, thank you!

Reply:
hey, I think you misunderstanding the docs just as I did a few weeks ago!
I am not sure about this BUT the convention is that you should use the filter and not use the JS global t function in JS! BUT you should prefer rails translations (with lazy load) to js translation. Correct @mkllnk ?

Reply:
imo, in this case I think you should keep the rails translation, not js.

Reply:
Sorry for the confusion. We have to improve the documentation somehow.

- This is a Rails view. So we should use Rails' `I18n.t` method:
```haml
%input.ofn-select2.fullwidth{ :id => 'involving_filter', type: 'number', blank: "{id: 0, name: '#{t(".any_enterprise")}'}", data: 'enterprises', ng: { model: 'involvingFilter' } }
```
- In Angular views, we should use the `t` filter.
- In other Angular/JS code where filters are not easily accessible, we should use Javascript's `I18n.t` function.

Reply:
I updated the wiki to reflect that we should still prefer the Ruby method in Rails views: ["Rails view translations: Prefer Ruby `t` method"](https://github.com/openfoodfoundation/openfoodnetwork/wiki/Internationalisation-(i18n)#rails-view-translations-prefer-ruby-t-method).

This PR has been updated as well. :slightly_smiling_face: This is still in "Code Review". Kindly just check again, then move to "Test Ready" when okay.